code/game/objects/items/defibrillator.dm

```diff
@@ -128,215 +136,193 @@
-	var/defib_heal_amt = damage_threshold
-
 	//job knowledge requirement
 	var/medical_skill = user.skills.getRating(SKILL_MEDICAL)
 	if(medical_skill < SKILL_MEDICAL_PRACTICED)
 		user.visible_message(span_notice("[user] fumbles around figuring out how to use [src]."),
 		span_notice("You fumble around figuring out how to use [src]."))
 		var/fumbling_time = SKILL_TASK_AVERAGE - (SKILL_TASK_VERY_EASY * medical_skill) // 3 seconds with medical medical_skill, 5 without
-		if(!do_after(user, fumbling_time, NONE, H, BUSY_ICON_UNSKILLED))
+		if(!do_after(user, fumbling_time, NONE, patient, BUSY_ICON_UNSKILLED))
 			return
 
-	defib_heal_amt = DEFIBRILLATOR_HEALING_TIMES_SKILL(medical_skill)
-
-	if(!ishuman(H))
-		to_chat(user, span_warning("You can't defibrilate [H]. You don't even know where to put the paddles!"))
-		return
-	if(!ready)
-		to_chat(user, span_warning("Take [src]'s paddles out first."))
-		return
-	if(dcell.charge <= charge_cost)
-		user.visible_message(span_warning("[icon2html(src, viewers(user))] \The [src] buzzes: Internal battery depleted. Cannot analyze nor administer shock."))
-		to_chat(user, maybe_message_recharge_hint())
-		return
-	if(H.stat != DEAD)
-		user.visible_message(span_warning("[icon2html(src, viewers(user))] \The [src] buzzes: Vital signs detected. Aborting."))
-		return
+	var/defib_heal_amt = DEFIBRILLATOR_HEALING_TIMES_SKILL(medical_skill, damage_threshold)
 
-	if((HAS_TRAIT(H, TRAIT_UNDEFIBBABLE ) && !issynth(H)) || H.suiciding) //synthetic species have no expiration date
-		user.visible_message(span_warning("[icon2html(src, viewers(user))] \The [src] buzzes: Patient is braindead. No remedy possible."))
+	if(dcell.charge <= charge_cost) // This is split from defib_ready because we don't want to check charge AFTER delivering shock
+		user.visible_message(span_warning("[icon2html(src, viewers(user))] \The [src] buzzes: Internal battery depleted. Seek recharger. Cannot analyze nor administer shock."))
+		to_chat(user, span_boldwarning("You can recharge the defibrillator by click-dragging it onto a corpsman backpack or satchel, or putting it in a recharger."))
 		return
 
-	if(!H.has_working_organs() && !(H.species.species_flags & ROBOTIC_LIMBS))
-		user.visible_message(span_warning("[icon2html(src, viewers(user))] \The [src] buzzes: Patient's organs are too damaged to sustain life. Deliver patient to a MD for surgical intervention."))
+	if(!defib_ready(patient, user))
 		return
 
-	if((H.wear_suit && H.wear_suit.atom_flags & CONDUCT))
-		user.visible_message(span_warning("[icon2html(src, viewers(user))] \The [src] buzzes: Paddles registering >100,000 ohms, Possible cause: Suit or Armor interferring."))
+	var/fail_reason
+	if(patient.check_defib() & (DEFIB_PREVENT_REVIVE_STATES))
+		// Special bit for preventing the do_after if they almost 100% can't come back.
+		// We'll check the rest of the check_defib states along with these after shocking, incase their status changes mid defib.
+		switch(patient.check_defib())
```

Comment:
Why are we running this proc twice in a row?

Reply:
This is what I could think of to retain the original behavior of being unable to waste defib charge on someone who won't be coming back.

Original defib code wouldn't let you even start the defib progress bar if they were undefibbable or whatever.

Reply:
No I mean, why are we literally running this proc twice in a row? There is nothing happening between these two calls, so the result is the same. So either switch straight away for save the result of the proc in a var.